nservicebus/messaging/delayed-delivery.md

```diff
@@ -14,11 +14,15 @@ The message doesn't have to be dispatched immediately after sending, it can be d
 NOTE: Only send operations can be deferred. Publish and reply operations cannot be deferred.
 
 ## Handling current message later
-To handle the current message at a later time `bus.HandleCurrentMessageLater()` method can be used.  This was the initial way of *deferring* a message. This would put the message at the end of the queue and the message eventually will be pick up again once other messages are processed. The problem with this approach is that if there are no other messages in the queue (or that many of them) the message will pop right back up so the intended deferring might not happen. 
+To handle the current message at a later time `bus.HandleCurrentMessageLater()` method can be used.  This was the initial way of *deferring* a message. This would create a copy of the message that has the same identifier, header and body, and then put the new message at the end of the queue. The message eventually will be pick up again once other messages in the queue are processed. 
 
-Note: RabbitMQ tries to put the message back in the queue as close to its original position as possible, which may mean at the head of the queue where it will be picked up immediately again. 
+There are two caveats with this method:
 
-A bigger potential issue with this method is that although the message is put back into the queue, the transaction(s) will still commit. This means if a database entity is mutated in the handler code and then the message is put back into the queue, the database transaction will still commit. As such, this method will be depricated in NServiceBus version 7.0 and it is recommended to use either [Delayed Retries](/nservicebus/recoverability/#delayed-retries) or one of the deferring mechanisms below, depending on the case at hand.  
+- As mentioned above a new copy of the message will be put back at the end of the queue. To make this work, the message pipeline will not abort which means any business transaction will also get comitted. 
+
+- If there are no other messages in the queue, when the message goes back into the queue it will be picked up immediately. This will have an effect of an endless loop which will be noted as high system resource utilization by the endpoint.
```

Comment:
Yes empty queue would make the loop worse but the key point is that your "exit criteria" needs to be rock solid to avoid the loop.

`if loadedCustomer == null` 

won't cut it